### PR TITLE
Add LocalSparkUtils implementation of msparkutils.credentials.getToken

### PR DESCRIPTION
**Pull request by MikeEvansLarah**

### CODE REVIEW

File: src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py [hunk 1]
```diff
@@ -62,6 +64,28 @@ def getSecretWithLS(self, linked_service: str, secret_name: str) -> str:
...
+            "AzureDataExplorer": "https://kusto.kusto.windows.net/.default",
+            "AzureOSSDB": "https://ossrdbms-aad.database.windows.net/.default",
+        }
+
+        scope = scopes.get(audience)
+
+        if not scope:
+            raise ValueError(f"Unsupported audience '{audience}'")
+
+        default_credential = DefaultAzureCredential()
```
Comment (edfreeman) [inline src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py:85]: Wonder if we should catch the inevitable "not logged in" exception and write a message explaining the specific way they should be logging in (i.e. via the az CLI using Device Code flow).

Comment (edfreeman) [inline src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py:85]: I also wonder whether we do need optional config to specify the target tenant ID?

File: entrypoints/example.ipynb [hunk 2]
```diff
@@ -22,6 +25,15 @@
     "print(cred)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "token = spark_utils.credentials.getToken(\"Synapse\")"
```
Comment (edfreeman) [inline entrypoints/example.ipynb:34]: You wanna add an `assert token` just to check it's non-empty (and therefore functional)?

File: src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py [hunk 3]
```diff
@@ -62,6 +64,39 @@ def getSecretWithLS(self, linked_service: str, secret_name: str) -> str:
...
+
+        tenant_id = None
+        get_token_config = self.config.get("getToken")
+        if get_token_config:
+            tenant_id = get_token_config.get("tenantId")
+
+        credential = AzureCliCredential()
+
+        try:
+            token = credential.get_token(scope, tenant_id=tenant_id)
```
Comment (edfreeman) [inline src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py:93]: This still works if the `tenant_id` is `None`?

Comment (MikeEvansLarah) [reply src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py:93]: Yup

**Review by edfreeman — APPROVED** [review]

**Pull request merged**